client/service/src/client/call_executor.rs

```diff
@@ -279,3 +319,72 @@ impl<B, E, Block> sp_version::GetRuntimeVersion<Block> for LocalCallExecutor<B,
 		CallExecutor::runtime_version(self, at).map_err(|e| format!("{:?}", e))
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use substrate_test_runtime_client::{LocalExecutor, GenesisInit, runtime};
+	use sc_executor::{NativeExecutor, WasmExecutionMethod};
+	use sp_core::{traits::{WrappedRuntimeCode, FetchRuntimeCode}, testing::TaskExecutor};
+	use sc_client_api::in_mem;
+	use sc_block_builder::BlockBuilderProvider;
+
+	#[test]
+	fn should_get_overwrite_if_exists() {
+		let executor =
+			NativeExecutor::<LocalExecutor>::new(WasmExecutionMethod::Interpreted, Some(128), 1);
+
+		let overwrites = crate::client::wasm_overwrite::dummy_overwrites(&executor);
+		let onchain_code = WrappedRuntimeCode(substrate_test_runtime::wasm_binary_unwrap().into());
+		let onchain_code = RuntimeCode {
+			code_fetcher: &onchain_code,
+			heap_pages: Some(128),
+			hash: vec![0, 0, 0, 0],
+		};
+
+		let backend = Arc::new(in_mem::Backend::<runtime::Block>::new());
+
+		// wasm_runtime_overwrites is `None` here because we construct the
+		// LocalCallExecutor directly later on, and the client is just used for the convenience
+		// of creating a block.
+		let client_config = ClientConfig {
+			offchain_worker_enabled: false,
+			offchain_indexing_api: false,
+			wasm_runtime_overwrites: None,
+		};
+
+		let client = substrate_test_runtime_client::client::new_with_backend::<
+			_,
+			_,
+			runtime::Block,
+			_,
+			runtime::RuntimeApi,
+		>(
+			backend.clone(),
+			executor.clone(),
+			&substrate_test_runtime_client::GenesisParameters::default().genesis_storage(),
+			None,
+			Box::new(TaskExecutor::new()),
+			None,
+			Default::default(),
+		).expect("Creates a client");
+
+		client.new_block(Default::default())
```

Comment:
If we don't build a block, the LocalCallExecutors `runtime_version` method will fail because it gets a block-by-id and then the state trie for that block to get the RuntimeCode before executing it for the version

Reply:
Interestingly, the test still passes if `client.new_block` is removed, but fails if instantiating the client is removed ...

I guess instantiating the test client auto-inserts a test block? TIL